server/src/main/scala/net/wiringbits/actions/CreateUserAction.scala

```diff
@@ -33,35 +41,57 @@ class CreateUserAction @Inject() (
       hmacToken = TokensHelper.doHMACSHA1(token.toString.getBytes(), userTokensConfig.hmacSecret)
 
       // create the user
-      createUser = repositories.models.User
-        .CreateUser(
-          id = UUID.randomUUID(),
-          name = request.name,
-          email = request.email,
-          hashedPassword = hashedPassword,
-          verifyEmailToken = hmacToken
-        )
-      _ <- usersRepository.create(createUser)
+      createUser <- createUser(request = request, hashedPassword = hashedPassword, hmacToken = hmacToken)
 
       // then, send the verification email
-      _ <- emailsHelper.sendRegistrationEmailWithVerificationToken(
-        User(
-          id = createUser.id,
-          name = request.name,
-          email = request.email,
-          hashedPassword = hashedPassword,
-          createdAt = Instant.now,
-          verifiedOn = None
-        ),
-        token
-      )
-    } yield CreateUser.Response(id = createUser.id, email = createUser.email, name = createUser.name)
+      _ <- emailsHelper.sendRegistrationEmailWithVerificationToken(createUser, token)
+    } yield CreateUser.Response(
+      id = createUser.userId.value.value,
+      email = request.email,
+      name = request.name
+    )
   }
 
   private def validations(request: CreateUser.Request) = {
     for {
       _ <- ValidateCaptcha(reCaptchaApi, request.captcha)
-      _ <- ValidateEmailIsAvailable(usersRepository, request.email)
+      _ <- ValidateEmailIsAvailable(request.email)
     } yield ()
   }
+
+  private def createUser(request: CreateUser.Request, hashedPassword: String, hmacToken: String): Future[UsersRow] = {
+    val createUser = UsersRow(
+      userId = UsersId(TypoUUID(UUID.randomUUID())),
+      name = request.name.string,
+      lastName = None,
+      email = TypoUnknownCitext(request.email.string),
+      password = hashedPassword,
+      createdAt = TypoOffsetDateTime(clock.instant().atOffset(ZoneOffset.UTC)),
+      verifiedOn = None
+    )
+
+    val createToken = UserTokensRow(
+      userTokenId = UserTokensId(TypoUUID(UUID.randomUUID())),
+      token = hmacToken,
+      tokenType = UserTokenType.EmailVerification.toString,
+      createdAt = TypoOffsetDateTime(clock.instant().atOffset(ZoneOffset.UTC)),
+      expiresAt = TypoOffsetDateTime(
+        clock.instant().atOffset(ZoneOffset.UTC).plus(userTokensConfig.emailVerificationExp.toHours, ChronoUnit.HOURS)
+      ),
+      userId = createUser.userId
+    )
+
+    val createUserLogs = UserLogsRow(
+      userLogId = UserLogsId(TypoUUID(UUID.randomUUID())),
+      userId = createUser.userId,
+      message = s"Account created, name = ${request.name}, email = ${request.email}",
+      createdAt = TypoOffsetDateTime(clock.instant().atOffset(ZoneOffset.UTC))
+    )
+
+    for {
+      user <- Future(UsersRepoImpl.insert(createUser))
```

Comment:
Different to our repository, this is likely to run each operation in a separate transaction, we need to keep our behavior